### PR TITLE
Setup associations

### PR DESCRIPTION
**Pull request by Pazzo97**
### In this PR I did the following:

- Created a class Classroom
- Created the has-many/belongs-to relationship between Classroom and Student
- Created a class Book
- Created a class Rental
- Created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_reader :rentals
+  attr_accessor :title, :author
+
+  def initialize(title, author)
+    @author = author
+    @title = title
+    @rentals = []
+  end
+end
```
Comment (VICTOR-CHUKWU) [inline book.rb]: - [ ] they should be a method for adding new rental the method should look like this :point_down: 
```
def add_rental(person, date)
    # *the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association. 
 ....
  end

```

File: person.rb [hunk 2]
```diff
@@ -2,13 +2,14 @@
...
+  attr_reader :id, :rentals
 
   def initialize(age, name = 'Unknown', parent_permission: true)
     @id = rand(1..10_000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
     super
   end
```
Comment (VICTOR-CHUKWU) [inline person.rb]: - [ ] they should also be a method for adding rentals in this class, kindly fix this
```ruby
def add_rental(book, date)
    # *The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association
    Rental.new(date, book, self)
  end
```

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @Pazzo97  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+class Book
+  attr_reader :rentals
+  attr_accessor :title, :author
```
Comment (BenMukebo) [inline book.rb]: - [OPTIONAL] Getters and Setters for @rentals should be present in the book 👇🏻 

 ```ruby
  attr_accessor :title, :author, :rentals
```

File: person.rb [hunk 4]
```diff
@@ -2,13 +2,14 @@
...
```
Comment (BenMukebo) [inline person.rb:15]: - [ ] As requested by the previous reviewer:

> - [ ] they should also be a method for adding rentals in this class, kindly fix this
> ```ruby
> def add_rental(book, date)
>     # *The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association
>     Rental.new(date, book, self)
>   end
> ```

- Kindly address this 👍🏻

**Review by BenMukebo — CHANGES_REQUESTED** [review]
Hi @Pazzo97, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

You have submitted a project, that Linters check are not running in the GitHub/action
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please Tag me on (@BenMukebo) when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by keneogu — APPROVED** [review]
Hi @Pazzo97,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**